core/services/ocr2/plugins/mercury/integration_test.go

```diff
@@ -73,6 +74,14 @@ func randomFeedID() [32]byte {
 func TestIntegration_Mercury(t *testing.T) {
 	t.Parallel()
 
+	t.Cleanup(func() {
+		// Sleep for a little to give time for wsrpc's client goroutines to pick up
+		// the connection closures and emit any final logs. Otherwise, there's a chance
+		// that the logs will be emitted after this test is completed, causing the
+		// test to panic.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
@samsondav I added this in because the integration test was consistently failing with `panic: Log in goroutine after TestIntegration_Mercury has completed: 2023-07-10T19:54:32.705Z	ERROR	oracle_mercury3.EVM.Mercury.WSRPCPool.WSRPC	transport/websocket_client.go:128	[wsrpc] Read error	{"version": "2.2.0@b860c2d", "mercuryServerURL": "127.0.0.1:44157", "err": "websocket: close 1000 (normal)"}`.

[Here](https://github.com/smartcontractkit/wsrpc/blob/main/internal/transport/websocket_client.go#L128) is where the log line in question is emitted. I don't see a way to deterministically wait for this inside the test, but let me know if you have have a better idea for how to handle this!

Reply:
Yup fully agree that it's not ideal. I could be wrong, but my take is that things are being closed properly, however because this log line happens in a separate client goroutine after it detects the conn has been closed, there's not a good way for the main test goroutine to wait for that client goroutine to wrap up, before the test ends.

Reply:
oookay I finally figured it out. You were right that something wasn't getting closed properly. I left out `persistenceManager.Start()` but not `Close()` so that close fails which in turn prevents closing the wsrpc client connections.

To make this more obvious in the future, I added a section to the integration test to ensure there are no DPANIC level logs being emitted. If there are, then the test will fail with something like this:
```
    integration_test.go:152: Found logs with DPANIC or higher level:
        2023-07-13T10:19:39+10:00	DPANIC	oracle_mercury3.JobSpawner	job/spawner.go:168	Error stopping job service
        2023-07-13T10:19:39+10:00	DPANIC	oracle_mercury3.JobSpawner	job/spawner.go:168	Error stopping job service
        2023-07-13T10:19:39+10:00	DPANIC	oracle_mercury3.JobSpawner	job/spawner.go:168	Error stopping job service
```